client.go

```diff
@@ -134,6 +162,49 @@ func (c *Client) Browse(req *ua.BrowseRequest) (*ua.BrowseResponse, error) {
 	return res, err
 }
 
+// GetEndpoints reads all available endpoints from the server at endpoint
+func GetEndpoints(endpoint string) ([]*ua.EndpointDescription, error) {
+	var res *ua.GetEndpointsResponse
+
+	ctx := context.Background()
+
+	c := NewClient(endpoint, nil)
+	conn, err := uacp.Dial(ctx, c.Addr)
+	if err != nil {
+		return nil, err
+	}
+	sechan := uasc.NewSecureChannel(conn, nil)
+	if err := sechan.Open(); err != nil {
+		conn.Close()
+		return nil, err
+	}
+	sechan.EndpointURL = c.Addr
+	c.sechan = sechan
+
+	reqHeader := &ua.RequestHeader{
+		AuthenticationToken: ua.NewTwoByteNodeID(0),
+		Timestamp:           time.Now(),
+		AdditionalHeader:    ua.NewExtensionObject(nil),
+	}
+
+	req := &ua.GetEndpointsRequest{
+		RequestHeader: reqHeader,
+		EndpointURL:   endpoint,
+	}
+
+	err = c.sechan.Send(req, func(v interface{}) error {
+		r, ok := v.(*ua.GetEndpointsResponse)
+		if !ok {
+			return fmt.Errorf("invalid response: %T", v)
+		}
+		res = r
+		return nil
+	})
+
+	c.Close()
```

Comment:
Usually, you call `defer Close()`with after `Open` like below. Wait until I've pushed the client refactor.

```go
c, err := Open(...)
if err != nil {
    return err
}
defer c.Close()
```